Sources/Utility/ArgumentParser.swift

```diff
@@ -459,17 +527,48 @@ public final class ArgumentParser {
         usage: String? = nil
     ) -> PositionalArgument<T> {
         precondition(subparsers.isEmpty, "Positional arguments are not supported with subparsers")
-        precondition(optional ? positionalArgs.count <= 1 : true,
-            "Only one positional argument is allowed if optional setting is enabled.")
-        optionalPositionalArg = optional
-        let arg = PositionalArgument<T>(name: positional, usage: usage)
-        positionalArgs.append(AnyArgument(arg))
-        return arg
+        precondition(canAcceptPositionalArguments, "Can not accept more positional arguments")
+        
+        if optional {
+            canAcceptPositionalArguments = false
+        }
+        
+        let argument = PositionalArgument<T>(name: positional, usage: usage)
+        positionalArguments.append(ParserPositionalArgument(
+            argument: AnyArgument(argument),
+            isOptional: optional,
+            strategy: .one))
+        return argument
+    }
+
+    /// Adds an argument to the parser.
+    /// Note: Only one multiple-value positional argument is allowed.
```

Comment:
Multiple will stop whenever it reaches a non-positional argument and allows us to have more than one multiple-value options:

`exe --multiple-option1 arg1 arg2 arg3 --multiple-option2 arg1 arg2 arg3`